src/swarm-scenario/Swarm/Game/Scenario/Topography/Structure/Assembly.hs

```diff
@@ -38,18 +40,21 @@ overlaySingleStructure ::
   Either Text (MergedStructure (Maybe a))
 overlaySingleStructure
   inheritedStrucDefs
-  (Placed p@(Placement _ _shouldTruncate pose@(Pose loc orientation)) ns)
+  (Placed p@(Placement _ shouldTruncate pose@(Pose loc orientation)) ns)
   (MergedStructure inputArea inputPlacements inputWaypoints) = do
     MergedStructure overlayArea overlayPlacements overlayWaypoints <-
       mergeStructures inheritedStrucDefs (WithParent p) $ structure ns
 
     let mergedWaypoints = inputWaypoints <> map (fmap $ placeOnArea overlayArea) overlayWaypoints
         mergedPlacements = inputPlacements <> map (placeOnArea overlayArea) overlayPlacements
-        mergedArea = overlayGrid inputArea pose overlayArea
+        mergedArea =
+          if shouldTruncate
+            then overlayGridTruncated (gridContent inputArea) pose overlayArea
+            else overlayGridExpanded (gridContent inputArea) pose overlayArea
```

Comment:
TODO: use condition to select the function name, then make a single application of that function